### PR TITLE
added a first package.xml

### PR DESCRIPTION
**Pull request by marc-hanheide**
added first `package.xml` adressing #15

This is also to test jenkins builds in https://lcas.lincoln.ac.uk/jenkins/job/pr-indigo-v4r/ and https://lcas.lincoln.ac.uk/jenkins/job/devel-indigo-v4r/

### CODE REVIEW

**Comment by marc-hanheide** [comment]
retest this please

**Comment by marc-hanheide** [comment]
@mzillich @ToMadoRe I'm guessing some of the dependencies for V4R here... you may want to add others... This PR starts to compile V4R as it is right now, but some things might still be missing

**Comment by marc-hanheide** [comment]
OK, @mzillich and @ToMadoRe there is another undocumented dependency: 

```
error: GL/glew.h: No such file or directory
 #include "GL/glew.h"
```

I suggest you checkout this branch "package_xml" and add all the `<build_depend>` that you know of and push this to this branch and then we should end up with a version that compiles at least? Make sure you give the correct rosdep key as a dependency and *not* the Ubuntu package name. E.g. it must be `suitespare` instead of `libsuitespare-dev`. The rosdep database is to be found at https://github.com/strands-project/rosdistro/blob/strands-devel/rosdep/base.yaml, e.g check [suitesparse](https://github.com/strands-project/rosdistro/blob/strands-devel/rosdep/base.yaml#L3031-L3036). Currently, `libglew-dev` seems to be missing, but there might be more, and best is you add those yourself. Also, you might want to checks in your `CMakeLists.txt`, as for instance `glew` is currently not checked for in there.

**Comment by marc-hanheide** [comment]
You might want to take @arbeitor off the list of maintainers in `package.xml` as I understand he left?

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  <maintainer email="faeulhammer@acin.tuwien.ac.at">Thomas Fäulhammer</maintainer>
+  <maintainer email="aldoma@acin.tuwien.ac.at">Aitor Aldoma</maintainer>
+  <maintainer email="zillich@acin.tuwien.ac.at">Michael Zillich</maintainer>
+
+
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <!-- Commonly used license strings: -->
+  <!--   BSD, MIT, Boost Software License, GPLv2, GPLv3, LGPLv2.1, LGPLv3 -->
+  <license>MIT</license>
```
Comment (marc-hanheide) [inline package.xml]: Also, please set the license to whatever license you deem correct.

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0"?>
+<package>
+  <name>v4r</name>
+  <version>1.0.0</version>
```
Comment (marc-hanheide) [inline package.xml]: not sure this is the correct version number. I took it from [v4r.pc.in](https://github.com/strands-project/v4r/blob/master/v4r.pc.in).

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <!-- Commonly used license strings: -->
+  <!--   BSD, MIT, Boost Software License, GPLv2, GPLv3, LGPLv2.1, LGPLv3 -->
+  <license>MIT</license>
+
+
+  <!-- Url tags are optional, but mutiple are allowed, one per tag -->
+  <!-- Optional attribute type can be: website, bugtracker, or repository -->
+  <!-- Example: -->
+  <url type="website">https://github.com/strands-project/v4r</url>
```
Comment (marc-hanheide) [inline package.xml]: Set this to your local V4R page?

**Comment by marc-hanheide** [comment]
Oh, why did you merge yet? Generally, we should adopt a strategy of not merging PRs that failed, but fix them first. For now, no problem of course.

**Comment by marc-hanheide** [comment]
You will have noticed that you don't have to trigger it, but with every push to an open PR or to the "master" branch it does it for you, and you, as maintainer, should get an email back.

**Comment by ToMadoRe** [comment]
okay, but since it has not been compiling yet anyway, I can just push straight away to the master branch, right?

**Comment by ToMadoRe** [comment]
So, how can I interpret the output of the build now?
https://lcas.lincoln.ac.uk/jenkins/job/devel-indigo-v4r/3/console
Is there something wrong with `freeglut3-dev`?

**Comment by marc-hanheide** [comment]
Yes, it's not a rosdep key `KeyError: 'freeglut3-dev'` As I said in https://github.com/strands-project/v4r/pull/16#issuecomment-125248030 it needs to be a rosdep key you put in `package.xml`

**Comment by ToMadoRe** [comment]
but what's the right name then, I thought I need to take this one:
https://github.com/strands-project/rosdistro/blob/strands-devel/rosdep/base.yaml#L808

**Comment by ToMadoRe** [comment]
is it just glut?

**Comment by marc-hanheide** [comment]
it's `glut`, the key, that can be resolved into different packages, depending on the system.

**Comment by ToMadoRe** [comment]
makes sense... I will update and try again ;-)

**Pull request merged**